### PR TITLE
Fix parsing of inline code blocks with multiple backticks

### PR DESCRIPTION
**Pull request by srawlins**
Fixes https://github.com/dart-lang/markdown/issues/259

### CODE REVIEW

**Comment by srawlins** [comment]
The failed internal check is fine; we have a broken patch file.

File: lib/src/block_parser.dart [hunk 1]
```diff
@@ -404,12 +406,28 @@ class CodeBlockSyntax extends BlockSyntax {
...
 class FencedCodeBlockSyntax extends BlockSyntax {
-  RegExp get pattern => _codePattern;
+  RegExp get pattern => _codeFencePattern;
 
   const FencedCodeBlockSyntax();
 
+  bool canParse(BlockParser parser) {
+    var match = pattern.firstMatch(parser.current);
+    if (match == null) return false;
+    final codeFence = match.group(1);
```
Comment (munificent) [inline lib/src/block_parser.dart]: Use `var` here and below or `final` above.

File: lib/src/block_parser.dart [hunk 2]
```diff
@@ -404,12 +406,28 @@ class CodeBlockSyntax extends BlockSyntax {
...
+    final infoString = match.group(2);
+    // From the CommonMark spec:
+    //
+    // > If the info string comes after a backtick fence, it may not contain
+    // > any backtick characters.
+    if (codeFence.codeUnitAt(0) == $backquote &&
```
Comment (munificent) [inline lib/src/block_parser.dart]: How about:

```dart
return codeFence.codeUnitAt(0) != $backquote ||
    !infoString.codeUnits.contains($backquote));
```

**Comment by kevmoo** [comment]
Land it!

**Pull request merged**